### PR TITLE
fix(field): Align horizontal Field labels with control text

### LINKED ISSUE #30470: [Bug]: horizontal field: label and values are not vertically aligned
### Library

React Northstar / v0 (@fluentui/react-northstar)

### System Info

```shell
System:
    OS: Windows 11 10.0.22631
    CPU: (32) x64 13th Gen Intel(R) Core(TM) i9-13900K
    Memory: 68.75 GB / 95.75 GB
  Browsers:
    Edge: Chromium (121.0.2277.98)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://codesandbox.io/p/sandbox/affectionate-beaver-8ddjk8?file=%2Fsrc%2FApp.js

## Actual Behavior
Label and value are not vertically aligned: The label is higher up than the value in the input.
This is visible on your webpage: https://react.fluentui.dev/?path=/docs/components-field--default#horizontal-orientation.
Just enter a value in the input, then scroll the content up until the label disappears: the value will still be partly visible.
The provided reproduction is not a real one - it is only so that this issue is not auto closed.
By the way: Repreductions are hard when you dont link to an existing template. And codesandbox seems to loose free usage.

## Expected Behavior
Label and values should vertically align.

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

Correcting the issue seems to be as simple as adding `align-items: center` to the Field grid.

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by behowell**
## Previous Behavior

Horizontal Field labels did not line up with the text of the Input or other control inside the field.

## New Behavior

Adjust the horizontal label padding to line up with the control's text.

## Related Issue(s)

- Fixes #30470

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30732/perf-test-react-components/Avatar-mount_base.html">642</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30732/perf-test-react-components/Avatar-mount.html">661</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30732/perf-test-react-components/Button-mount_base.html">293</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30732/perf-test-react-components/Button-mount.html">296</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30732/perf-test-react-components/Field-mount_base.html">1151</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30732/perf-test-react-components/Field-mount.html">1102</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30732/perf-test-react-components/FluentProvider-mount_base.html">729</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30732/perf-test-react-components/FluentProvider-mount.html">720</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30732/perf-test-react-components/FluentProviderWithTheme-mount_base.html">78</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30732/perf-test-react-components/FluentProviderWithTheme-mount.html">90</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30732/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">70</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30732/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">68</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30732/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">86</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30732/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">77</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30732/perf-test-react-components/MakeStyles-mount_base.html">849</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30732/perf-test-react-components/MakeStyles-mount.html">853</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30732/perf-test-react-components/Persona-mount_base.html">1739</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30732/perf-test-react-components/Persona-mount.html">1719</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30732/perf-test-react-components/SpinButton-mount_base.html">1339</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30732/perf-test-react-components/SpinButton-mount.html">1378</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports | Baseline (minified/GZIP) | PR    | Change     |
| :---------------- | -----------------------: | ----: | ---------: |
| <samp>react-field</samp> <br /> <abbr title='bundle-size/Field.fixture.js'>Field</abbr> | `22.542 kB`<br />`8.609 kB` | `23.016 kB`<br />`8.721 kB` | `474 B` <br />`112 B` |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports | Size (minified/GZIP) |
| ----------------- | -------------------: |
| <samp>react-checkbox</samp> <br /> <abbr title='bundle-size/Checkbox.fixture.js'>Checkbox</abbr> | `35.656 kB`<br />`12.07 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr> | `102.686 kB`<br />`33.174 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr> | `104.06 kB`<br />`33.101 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `71.098 kB`<br />`20.515 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `217.205 kB`<br />`61.941 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `43.585 kB`<br />`14.352 kB` |
| <samp>react-datepicker-compat</samp> <br /> <abbr title='bundle-size/DatePicker.fixture.js'>DatePicker Compat</abbr> | `224.523 kB`<br />`62.94 kB` |
| <samp>react-input</samp> <br /> <abbr title='bundle-size/Input.fixture.js'>Input</abbr> | `26.891 kB`<br />`9.12 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr> | `7.944 kB`<br />`2.588 kB` |
| <samp>react-progress</samp> <br /> <abbr title='bundle-size/ProgressBar.fixture.js'>ProgressBar</abbr> | `17.428 kB`<br />`6.899 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/Radio.fixture.js'>Radio</abbr> | `32.95 kB`<br />`10.252 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/RadioGroup.fixture.js'>RadioGroup</abbr> | `15.354 kB`<br />`6.265 kB` |
| <samp>react-select</samp> <br /> <abbr title='bundle-size/Select.fixture.js'>Select</abbr> | `28.609 kB`<br />`10.204 kB` |
| <samp>react-slider</samp> <br /> <abbr title='bundle-size/Slider.fixture.js'>Slider</abbr> | `39.949 kB`<br />`12.968 kB` |
| <samp>react-spinbutton</samp> <br /> <abbr title='bundle-size/SpinButton.fixture.js'>SpinButton</abbr> | `36.774 kB`<br />`11.789 kB` |
| <samp>react-switch</samp> <br /> <abbr title='bundle-size/Switch.fixture.js'>Switch</abbr> | `35.14 kB`<br />`11.199 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/DataGrid.fixture.js'>DataGrid</abbr> | `165.189 kB`<br />`45.907 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/PrimitivesOnly.fixture.js'>Table (Primitives only)</abbr> | `45.324 kB`<br />`14.116 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableAsDataGrid.fixture.js'>Table as DataGrid</abbr> | `136.573 kB`<br />`36.817 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSelectionOnly.fixture.js'>Table (Selection only)</abbr> | `76.334 kB`<br />`20.553 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSortOnly.fixture.js'>Table (Sort only)</abbr> | `74.977 kB`<br />`20.155 kB` |
| <samp>react-textarea</samp> <br /> <abbr title='bundle-size/Textarea.fixture.js'>Textarea</abbr> | `30.947 kB`<br />`10.477 kB` |
| <samp>react-timepicker-compat</samp> <br /> <abbr title='bundle-size/TimePicker.fixture.js'>TimePicker</abbr> | `104.707 kB`<br />`34.558 kB` |
</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/77527ac13d9b9ffc627342ef70f2a417a8f18971'>77527ac13d9b9ffc627342ef70f2a417a8f18971</a></sub>
<div id="bundleSizeComment9423"/>

File: change/@fluentui-react-field-efd4dcce-2907-45c5-9380-899c87834d57.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-field-efd4dcce-2907-45c5-9380-899c87834d57.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=30732&commitId=0edac6eb37169f65c0ea1ac7284e023e3d8465a7&env=prod&clientType=FLUENTUI)


  <details><summary><b>Field</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Field.horizontal.default.chromium.png|311| Changed|
|Field.horizontal+longLabel.default.chromium.png|5132| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

Comment (spmonahan) [reply change/@fluentui-react-field-efd4dcce-2907-45c5-9380-899c87834d57.json:1]: These look good to me but I cannot approve them in the tool

**Review by spmonahan — APPROVED** [review]

File: change/@fluentui-react-field-efd4dcce-2907-45c5-9380-899c87834d57.json (hunk 1 above)
Comment (behowell) [reply change/@fluentui-react-field-efd4dcce-2907-45c5-9380-899c87834d57.json:1]: Updated to include more horizontal VR tests with different controls

**Review by mltejera — APPROVED** [review]
Checked the new VR tests ✅🎉

**Pull request merged**